### PR TITLE
Basic LVM VDO support

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: blivet/devices/lvm.py [hunk 1]
```diff
@@ -1789,8 +1789,132 @@ def populate_ksdata(self, data):
         data.pool_name = self.pool.lvname
 
 
+class LVMVDOPoolMixin(object):
```
Comment (dwlehman) [inline blivet/devices/lvm.py:1792]: Is it not useful/possible to derive from `LVMThinPoolMixin` to save some of the boilerplate?

File: blivet/devices/lvm.py [hunk 2]
```diff
@@ -1789,8 +1789,132 @@ def populate_ksdata(self, data):
...
+    def direct(self):
+        """ Is this device directly accessible? """
+        return False
+
+    def _create(self):
+        """ Create the device. """
+        raise NotImplementedError
+
+
+class LVMVDOLogicalVolumeMixin(object):
```
Comment (dwlehman) [inline blivet/devices/lvm.py:1843]: How similar is this to `LVMThinLogicalVolumeMixin`? A little weird, but aren't they mostly the same?

File: blivet/devices/lvm.py [hunk 3]
```diff
@@ -2066,6 +2196,11 @@ def readonly(self):
     def display_lv_name(self):
         return self.lvname
 
+    @property
+    @type_specific
+    def pool(self):
```
Comment (dwlehman) [inline blivet/devices/lvm.py:2201]: Why does the lvm-vdo lv need this when the thinp lv doesn't?

File: blivet/devices/lvm.py [hunk 4]
```diff
@@ -2167,6 +2302,19 @@ def resize(self):
...
+
+    @type_specific
+    def _remove_log_vol(self, lv):
+        pass
+
+    @property
+    @type_specific
+    def lvs(self):
+        return []
+
```
Comment (dwlehman) [inline blivet/devices/lvm.py:2317]: Same here, more or less... why is this needed if it wasn't needed for thinp?

Comment (vojtechtrefny) [reply blivet/devices/lvm.py:2317]: Because before this thinpool was the only "type" that needed these properties so these could be part of the thinpool "interface". Now we have two types that need it, so these need to be part of the mixin interface.

File: blivet/devices/lvm.py (hunk 2 above)
Comment (vojtechtrefny) [reply blivet/devices/lvm.py:1843]: The functions and properties are similar but usually differ a little. And the mixins are already very complicated and I'd like to avoid having `LVMVDOLogicalVolumeMixin` inheriting from `LVMThinLogicalVolumeMixin` and `LVMLogicalVolumeDevice` depending on both. This could go wrong very fast and complex dependencies in different versions of Python are huge mess.

I was thinking about adding another class, something like `LVMThinVolumeBase` (and `LVMPoolBase` for thinpool and vdopool) but I think this would make the code even less readable than now.

And also these two will differ more when we add support for creating VDO LVs/pools.
